registry/client/src/appRoutes/Show.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Show,
+    TabbedForm,
+    FormTab,
+    NumberInput,
+    TextInput,
+    BooleanInput,
+    required,
+    TextField,
+    ReferenceInput,
+    SelectInput,
+    ArrayInput,
+    SimpleFormIterator,
+    AutocompleteInput,
+    FormDataConsumer,
+} from 'react-admin';
+import JsonField from "../JsonField"; // eslint-disable-line import/no-unresolved
+import Help from "../components/Help";
+import Localization from "../Localization";
+import dataProvider from '../dataProvider';
+import Title from './Title';
+import { JSON_FIELD_VIEW_MODE } from '../constants';
+import { ShowTopToolbar } from '../components';
+
+const requiredSpecial = (value, allValues, props) => {
+    if (!allValues.specialRole) {
+        return undefined;
+    }
+
+    return required()(value, allValues);
+};
+
+const allowedAppKinds = [
+    { id: 'primary', name: 'Primary' },
+    { id: 'essential', name: 'Essential' },
+    { id: 'regular', name: 'Regular' },
+];
+
+const allowedSpecialRoles = [
+    { id: '404', name: '404' },
+];
+
+export default ({ permissions, hasList, hasEdit, hasShow, hasCreate, ...props }) => {
+    const [isMultiDomain, setIsMultiDomain] = useState(false);
+
+    useEffect(() => {
+        dataProvider.getList('router_domains', { pagination: false, sort: false, filter: false })
+            .then(({ data }) => {
+                if (!data.length) {
+                    return;
+                }
+
+                setIsMultiDomain(true);
+            });
+    }, []);
+
+    return (
+        <Show title={<Title />} {...props} actions={<ShowTopToolbar />}>
+            <TabbedForm {...props} toolbar={null}>
```

Comment:
Don't see a reason to use From inside Show component

Reply:
Also I think we need to use [Field Components](https://marmelab.com/react-admin/Fields.html) instead of [Input Components](https://marmelab.com/react-admin/Inputs.html) in Show views

Reply:
done